charts/dbildungscloud-iam-keycloak/templates/deployment.yaml

```diff
@@ -1,50 +1,57 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Release.Name }}-deployment
+  name: {{ template "common.names.deployment" . }}
+  namespace: {{ .Values.namespace }}
   labels:
-    app.kubernetes.io/name: dbildungs-iam
+    app.kubernetes.io/name: {{ .Values.commonLabels.name }}
 spec:
   selector:
     matchLabels:
-      layer: dbildungs-iam-keycloak
-  replicas: 1
+      app.kubernetes.io/layer: {{ .Values.selector.layer }}
```

Comment:
What is this label for and why is a selector defined in the deployment?

Reply:
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/